lib/3scale/backend/async_redis/client.rb

```diff
@@ -68,16 +62,38 @@ def connect_unix(opts)
             path = opts[:path]
 
             credentials = [opts[:username], opts[:password]]
-            protocol = Async::Redis::Protocol::RESP2
-            protocol = Async::Redis::Protocol::Authenticated.new(credentials, protocol) if credentials.any?
+            credentials = nil unless credentials.any?
             limit = opts[:max_connections]
 
             if opts.key? :sentinels
               raise InvalidURI.new(path, 'unix paths are not supported for sentinels')
             else
-              endpoint = EndpointHelpers.prepare_endpoint(path: path, ssl: opts[:ssl], ssl_params: opts[:ssl_params])
-              Async::Redis::Client.new(endpoint, protocol:, limit:)
+              endpoint = Async::Redis::Endpoint.unix(path, credentials:)
+              Async::Redis::Client.new(endpoint, limit:)
+            end
+          end
+
+          def create_ssl_context(ssl: false, ssl_params: nil)
+            if ssl
+              ssl_context = OpenSSL::SSL::SSLContext.new
+              if ssl_params
```

Comment:
There is also `:ca_file` params. I assume it doesn't need processing because it comes as a path, and it further passed to the library as a path?

Reply:
Yes, exactly, `:ca_file` is a string containing a path